### PR TITLE
Add user-agent header

### PR DESCRIPTION
**Pull request by ABevier**
I created a useFetch hook that wraps the Deno `fetch` function and automatically adds a `User-Agent` header to the request.

The header looks like `User-Agent: tea.cli/0.24.9`

### CODE REVIEW

**Review by mxcl — APPROVED** [review]

File: src/hooks/useFetch.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { useVersion } from "hooks";
+
+// useFetch wraps the native Deno fetch api and inserts a User-Agent header
+export default function useFetch(input: string | URL | Request, init?: RequestInit | undefined): Promise<Response> {
+    const requestInit = init ?? {} as RequestInit
+    requestInit.headers = { ...requestInit.headers, "User-Agent": `tea-cli/${useVersion()}` }
```
Comment (mxcl) [inline src/hooks/useFetch.ts:6]: Can we go with `tea/cli/version` or is that violating expectations for parsers of the user-agent?

Maybe `tea (cli)` is more typical.

Either way I'm not fond of the dash here.

Comment (ABevier) [reply src/hooks/useFetch.ts:6]: I think `tea(cli)/version` is probably fine if a bit atypical.  

You can see a bunch of examples here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Comment (mxcl) [reply src/hooks/useFetch.ts]: yeah you're right. Parenthesis are used for platform info not variants.

yeah, back to `tea-cli/version` I agree with your original assessment lol

Comment (mxcl) [reply src/hooks/useFetch.ts]: Or perhaps `tea.cli/version` which we've used elsewhere sometimes.

Comment (ABevier) [reply src/hooks/useFetch.ts]: tea.cli it is

**Pull request merged**